src/mps/mpo.jl

```diff
@@ -159,7 +167,7 @@ end
 function show(io::IO, W::MPO)
   print(io,"MPO")
   (length(W) > 0) && print(io,"\n")
-  @inbounds for (i, w) ∈ enumerate(inds.(W.A_))
+  @inbounds for (i, w) ∈ enumerate(inds.(tensors(W)))
```

Comment:
Hm does `inds.` here allocate a new array just to iterate? I'm sure it's not a real bottleneck or anything but it just seems like a thing we should avoid doing unless necessary. (I.e. a "premature pessimization".)

Reply:
Should we instead enumerate the tensors, and within the loop get their indices?

Reply:
Yes, I think that would be better. That is to do something like:
```
for (i,A) in enumerate(tensors(M))
```
then inside the loop body use `inds(A)` to grab just the indices of A.

Reply:
Awesome, on it!